src/main/java/reposense/util/FileUtil.java

```diff
@@ -122,8 +126,67 @@ public static void copyDirectoryFiles(Path src, Path dest) throws IOException {
         }
     }
 
+    /**
+     * Generates a list of Path contained in the given {@code directoryName} directory.
+     */
+    public static List<Path> listJsonFiles(String directoryName) throws IOException {
+        Path directory = Paths.get(directoryName);
+
+        List<Path> allJsonFiles = Files.walk(directory)
+                .filter(p -> p.toString().endsWith(".json"))
+                .distinct()
+                .collect(Collectors.toList());
+
+        return allJsonFiles;
+    }
+
+    /**
+     * Zips all the JSON files contained in the {@directoryPath} directory.
+     */
+    public static void zip(String directoryPath) throws IOException {
+
+        String zipFileName = "archiveJSON.zip";
+
+        List<Path> allJsonFiles = listJsonFiles(directoryPath);
+
+        try {
+            //byte buffer for I/O
+            byte[] buffer = new byte[1024];
+            int length;
+            FileOutputStream fos = new FileOutputStream(directoryPath + File.separator + zipFileName);
+            ZipOutputStream zos = new ZipOutputStream(fos);
+            for (int i = 0; i < allJsonFiles.size(); i++) {
+                FileInputStream fis = new FileInputStream(allJsonFiles.get(i).toFile());
+
+                // begin writing a new ZIP entry, positions the stream to the start of the entry data
+                if (isSummaryJson(allJsonFiles.get(i))) {
+                    zos.putNextEntry(new ZipEntry(allJsonFiles.get(i).getFileName().toString()));
+                } else {
+                    zos.putNextEntry(new ZipEntry(allJsonFiles.get(i).getParent().getFileName()
+                            + File.separator + allJsonFiles.get(i).getFileName()));
+                }
+
+                while ((length = fis.read(buffer)) > 0) {
+                    zos.write(buffer, 0, length);
+                }
+                zos.closeEntry();
+                fis.close();
+            }
+
+            zos.close();
```

Comment:
Its unsafe to close streams here, do it in ```finally``` block or use a ```try-with-resource``` block.

Read: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html